### PR TITLE
Version update for external libraries and bumping up versions for next release

### LINKED ISSUE #5743: [Security, Reliability] Upgrade all dependency libraries to latest stable version before GA
We have a few external dependencies like Reactor, OkHttp, Jackson, Proton-J etc. Before GA, upgrade to the latest release version of these dependencies.

### PR DESCRIPTION
**Pull request by srnagar**
### azure-core

| Dependency              | Current version | Latest version |
| ----------------------- | --------------- | -------------- |
| jackson-dataformat-xml  | 2.9.9           | 2.10.0         |
| jackson-datatype-jsr310 | 2.9.9           | 2.10.0         |
| jsr305                  | 3.0.2           | 3.0.2          |
| reactor-core            | 3.3.0.RELEASE   | 3.3.0.RELEASE  |
| slf4j-api               | 1.7.28          | 1.7.28         |

### azure-core-amqp

| Dependency               | Current version | Latest version |
| ------------------------ | --------------- | -------------- |
| proton-j                 | 0.31.0          | 0.33.2         |
| qpid-proton-j-extensions | 1.2.1           | 1.2.1          |

### azure-core-http-netty

| Dependency          | Current version | Latest version |
| ------------------- | --------------- | -------------- |
| netty-buffer        | 4.1.39.Final    | 4.1.42.Final   |
| netty-codec-http    | 4.1.39.Final    | 4.1.42.Final   |
| netty-handler       | 4.1.39.Final    | 4.1.42.Final   |
| netty-handler-proxy | 4.1.39.Final    | 4.1.42.Final   |
| reactor-netty       | 0.9.0.RELEASE   | 0.9.0.RELEASE  |

### azure-core-http-okhttp

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| okhttp     | 4.1.0           | 4.2.2          |

### azure-core-test

| Dependency   | Current version | Latest version |
| ------------ | --------------- | -------------- |
| junit        | 4.13-beta-3     | 4.13-beta-3    |
| reactor-test | 3.2.3.RELEASE   | 3.3.0.RELEASE  |
| slf4j-api    | 1.7.28          | 1.7.28         |

### azure-data-appconfiguration

No external dependencies.

### azure-messaging-eventhubs

No external dependencies.

### azure-message-eventhubs-checkpointstore-blob

No external dependencies.

### azure-identity

| Dependency      | Current version | Latest version |
| --------------- | --------------- | -------------- |
| msal4j          | 0.5.0-preview   | 0.5.0-preview  |
| oauth2-oidc-sdk | 6.13            | 6.14           |
| org.nanohttpd   | 2.3.1           | 2.3.1          |
| jna-platform    | 4.1.0           | 5.4.0          |

### azure-keyvault-certificates

| Dependency    | Current version | Latest version |
| ------------- | --------------- | -------------- |
| commons-codec | 1.10            | 1.13           |

### azure-keyvault-keys

| Dependency    | Current version | Latest version |
| ------------- | --------------- | -------------- |
| commons-codec | 1.10            | 1.13           |

### azure-keyvault-secrets

| Dependency    | Current version | Latest version |
| ------------- | --------------- | -------------- |
| commons-codec | 1.10            | 1.13           |

### azure-storage-blob

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| slf4j-api  | 1.7.28          | 1.7.28         |
| jsr305     | 3.0.2           | 3.0.2          |

### azure-storage-blob-batch

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| slf4j-api  | 1.7.28          | 1.7.28         |
| jsr305     | 3.0.2           | 3.0.2          |

### azure-storage-blob-cryptography

No external dependencies.

### azure-storage-common

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| slf4j-api  | 1.7.28          | 1.7.28         |
| jsr305     | 3.0.2           | 3.0.2          |

### azure-storage-file

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| jsr305     | 3.0.2           | 3.0.2          |

### azure-storage-file-datalake

| Dependency | Current version | Latest version |
| ---------- | --------------- | -------------- |
| jsr305     | 3.0.2           | 3.0.2          |

### azure-storage-queue

No external dependencies.

### azure-storage-queue-cryptography

No external dependencies.

Fixes #5743

### CODE REVIEW

**Comment by JonathanGiles** [comment]
@g2vinay why does key vault need commons codec?

File: sdk/tracing/azure-core-tracing-opencensus/pom.xml [hunk 1]
```diff
@@ -6,13 +6,13 @@
...
     <artifactId>azure-client-sdk-parent</artifactId>
-    <version>1.5.0</version>
+    <version>1.6.0</version>
     <relativePath>../../../pom.client.xml</relativePath>
   </parent>
 
   <groupId>com.azure</groupId>
   <artifactId>azure-core-tracing-opencensus</artifactId>
-  <version>1.0.0-preview.3</version>
+  <version>1.0.0-preview.4</version>
```
Comment (samvaity) [inline sdk/tracing/azure-core-tracing-opencensus/pom.xml]: wait for this to be released, before merging.

Comment (srnagar) [reply sdk/tracing/azure-core-tracing-opencensus/pom.xml]: Yes, I will wait until all releases are done before merging this.

Comment (srnagar) [reply sdk/tracing/azure-core-tracing-opencensus/pom.xml]: Yes, I will wait until all releases are done.

File: parent/pom.xml [hunk 2]
```diff
@@ -115,8 +115,8 @@
     <slf4j-log4j12.version>1.7.0</slf4j-log4j12.version>
     <reactor-core.version>3.3.0.RELEASE</reactor-core.version>
     <reactor-netty.version>0.9.0.RELEASE</reactor-netty.version>
-    <netty.version>4.1.39.Final</netty.version>
-    <okhttp.version>4.1.0</okhttp.version>
+    <netty.version>4.1.42.Final</netty.version>
```
Comment (anuchandy) [inline parent/pom.xml]: The version of reactor-core we're using is 0.9.0-RELEASE which depends on 4.1.39.Final version of Netty. I'm not sure about any issue with we using a version of netty different than reactor-core depends on.

Comment (srnagar) [reply parent/pom.xml]: It's a minor version upgrade and their release notes mention that 41, 42 are bug-fix releases with performance improvements. 

There's at least [one bug](https://github.com/Azure/azure-sdk-for-java/issues/5170) that was reported by a customer and fixed in `41.Final` release . So, I would prefer to upgrade unless we find issues with `42.Final` that's not in `39.Final`.

https://netty.io/news/2019/09/12/4-1-41-Final.html
https://netty.io/news/2019/09/25/4-1-42-Final.html

**Comment by weshaggard** [comment]
cc @JimSuplizio

**Comment by g2vinay** [comment]
Looks fine.
for KV, I'll be removing commons-codec before GA.

> @g2vinay why does key vault need commons codec?

Currently, we are using Base64  from commons-codec for encode/decode puproses.
But, yeah, I have some changes to port it to use Java Util Base 64.
With that added to master, we can get rid of commons-codec.

**Comment by srnagar** [comment]
Closing this PR in favor of https://github.com/Azure/azure-sdk-for-java/pull/5795.

@JimSuplizio is working on a tool to upgrade client library versions. So, the other PR will focus on upgrading only the external dependency versions.

**Pull request closed without merging**